File: frontend/scripts/sync-brand-public.mjs
/**
 * Copy brand/logos/product/{svg,png,icons} → frontend/public/brand for Next.js.
 * Same behavior as scripts/sync-brand-public.py (Node so Docker/CI need no Python).
 */
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ROOT = path.resolve(__dirname, "../..");
const SRC = path.join(ROOT, "brand", "logos", "product");
const DST = path.join(ROOT, "frontend", "public", "brand");

function copyRecursive(src, dst) {
  const stat = fs.statSync(src);
  if (stat.isDirectory()) {
    fs.mkdirSync(dst, { recursive: true });
    for (const name of fs.readdirSync(src)) {
      copyRecursive(path.join(src, name), path.join(dst, name));
    }
    return;
  }
  fs.mkdirSync(path.dirname(dst), { recursive: true });
  fs.copyFileSync(src, dst);
}

if (!fs.existsSync(SRC) || !fs.statSync(SRC).isDirectory()) {
  console.error(`Missing source directory: ${SRC}`);
  process.exit(1);
}

fs.mkdirSync(DST, { recursive: true });
for (const name of fs.readdirSync(DST)) {
  fs.rmSync(path.join(DST, name), { recursive: true, force: true });
}
for (const name of fs.readdirSync(SRC)) {
  const from = path.join(SRC, name);
  const to = path.join(DST, name);
  copyRecursive(from, to);
  console.log(`synced ${name}`);
}
console.log(`ok ${SRC} -> ${DST}`);

File: scripts/sync-brand-public.py
"""Copy brand/logos/product/{svg,png,icons} → frontend/public/brand for Next.js.

Prefer frontend/scripts/sync-brand-public.mjs (used by npm run sync-brand).
"""

from __future__ import annotations

import shutil
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
SRC = ROOT / "brand" / "logos" / "product"
DST = ROOT / "frontend" / "public" / "brand"


def main() -> None:
    if not SRC.is_dir():
        raise SystemExit(f"Missing source directory: {SRC}")
    DST.mkdir(parents=True, exist_ok=True)
    for path in DST.iterdir():
        if path.is_file():
            path.unlink()
        elif path.is_dir():
            shutil.rmtree(path)
    for path in SRC.iterdir():
        target = DST / path.name
        if path.is_file():
            shutil.copy2(path, target)
        elif path.is_dir():
            shutil.copytree(path, target)
        print(f"synced {path.name}")
    print(f"ok {SRC} -> {DST}")


if __name__ == "__main__":
    main()
